crates/predicate/src/verifiers/mod.rs

```diff
@@ -0,0 +1,82 @@
+//! Predicate type implementations.
+//!
+//! This module contains the individual implementations for each predicate type,
+//! organized in separate files for better maintainability.
+
+pub(crate) mod schnorr;
+pub(crate) mod sp1_groth16;
+
+pub(crate) use schnorr::SchnorrVerifier;
+pub(crate) use sp1_groth16::Sp1Groth16VerifierImpl;
+
+use crate::constants::{
+    ALWAYS_ACCEPT_PREDICATE_TYPE, BIP340_SCHNORR_PREDICATE_TYPE, NEVER_ACCEPT_PREDICATE_TYPE,
+    PredicateType, SP1_GROTH16_PREDICATE_TYPE,
+};
+use crate::errors::PredicateError;
+use crate::verifier::PredicateVerifier;
+
+/// Enum representing all supported predicate types and their implementations.
+///
+/// This enum serves as a dispatch mechanism to route verification calls to the
+/// appropriate predicate-specific implementation based on the predicate type.
+/// Each variant contains the implementation for a specific predicate backend.
+#[derive(Debug)]
+pub(crate) enum PredicateImpl {
+    /// Never accept predicate implementation (always fails verification).
+    NeverAccept,
+    /// Always accept predicate implementation (for testing and placeholders).
+    AlwaysAccept,
```

Comment:
This is dangerous, should be featured flag under `#[cfg(test)]`

Reply:
It's not for testing *the library*, it's for testing things that use the predicate spec.